ekncontent/ekncontent/eknc-domain.c

```diff
@@ -294,24 +316,39 @@ eknc_domain_initable_init (GInitable *initable,
   EkncDomain *self = EKNC_DOMAIN (initable);
   GError *local_error = NULL;
 
-  if (self->app_id == NULL || *self->app_id == '\0')
+  gboolean has_app_id = (self->app_id != NULL && *self->app_id != '\0');
+  gboolean has_path = (self->path != NULL && *self->path != '\0');
+
+  g_autoptr(GFile) subscriptions_dir = NULL;
+  g_autoptr(GFile) subscription_dir = NULL;
+  g_autoptr(GFile) bundle_subscriptions_dir = NULL;
+  g_autoptr(GFile) bundle_dir = NULL;
+
+  if (has_path)
+    {
+      subscription_dir = g_file_new_for_path (self->path);
```

Comment:
I just noticed this will be created if it does not exists which is something we do not really want in this case (user can give a wrong path)

you can use g_file_query_exists() to check and bail if the directory does not exists